data-service/src/test/java/com/amazonaws/blox/dataservice/repository/EnvironmentRepositoryDDBTest.java

```diff
@@ -198,4 +255,197 @@ public void geEnvironment() throws StorageException, EnvironmentNotFoundExceptio
+        environmentTargetVersionDDBRecordCaptor.getValue().getEnvironmentId());
+    assertEquals(
+        environmentVersion.getEnvironmentVersion(),
+        environmentTargetVersionDDBRecordCaptor.getValue().getEnvironmentVersion());
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void getEnvironmentTargetVersionNullEnvironmentId() throws StorageException {
+    environmentRepositoryDDB.getEnvironmentTargetVersion(null);
+  }
+
+  @Test(expected = StorageException.class)
+  public void getEnvironmentTargetVersionStorageException() throws StorageException {
+    doThrow(new AmazonServiceException(""))
+        .when(dynamoDBMapper)
+        .load(isA(EnvironmentTargetVersionDDBRecord.class));
+    environmentRepositoryDDB.getEnvironmentTargetVersion(environmentVersion.getEnvironmentId());
+  }
+
+  @Test
+  public void getEnvironmentTargetVersion() throws StorageException {
+    when(dynamoDBMapper.load(isA(EnvironmentTargetVersionDDBRecord.class)))
+        .thenReturn(environmentTargetVersionDDBRecord);
+
+    final EnvironmentVersion environmentVersionResult =
+        environmentRepositoryDDB.getEnvironmentTargetVersion(environmentVersion.getEnvironmentId());
+    assertEquals(environmentVersion, environmentVersionResult);
+
+    verify(dynamoDBMapper)
+        .load(EnvironmentTargetVersionDDBRecord.withHashKey(environmentVersion.getEnvironmentId()));
+  }
+
+  @Test(expected = StorageException.class)
+  public void listClustersStorageException() throws StorageException {
+    when(dynamoDBMapper.scan(
+            eq(EnvironmentTargetVersionDDBRecord.class), isA(DynamoDBScanExpression.class)))
+        .thenThrow(new AmazonServiceException(""));
+
+    environmentRepositoryDDB.listClusters();
+  }
+
+  @Test
+  public void listClusters() throws StorageException {
+    List<EnvironmentTargetVersionDDBRecord> environmentTargetVersionDDBRecords = new ArrayList<>();
+    environmentTargetVersionDDBRecords.add(environmentTargetVersionDDBRecord);
+    final EnvironmentTargetVersionDDBRecord secondRecord =
+        EnvironmentTargetVersionDDBRecord.builder()
+            .environmentId(SECOND_ENVIRONMENT_ID)
+            .environmentVersion(UUID.randomUUID().toString())
+            .cluster(SECOND_CLUSTER)
+            .build();
+    environmentTargetVersionDDBRecords.add(secondRecord);
+
+    final EnvironmentTargetVersionDDBRecord thirdRecordSameCluster =
+        EnvironmentTargetVersionDDBRecord.builder()
+            .environmentId(SECOND_ENVIRONMENT_ID)
+            .environmentVersion(UUID.randomUUID().toString())
+            .cluster(SECOND_CLUSTER)
+            .build();
+    environmentTargetVersionDDBRecords.add(thirdRecordSameCluster);
+
+    when(environmentTargetVersionDDBRecordPaginatedScanList.stream())
+        .thenReturn(environmentTargetVersionDDBRecords.stream());
+
+    when(dynamoDBMapper.scan(
+            eq(EnvironmentTargetVersionDDBRecord.class), isA(DynamoDBScanExpression.class)))
```

Comment:
The `isA(DynamoDBScanExpression.class)` is causing an `unchecked assignment` warning because it's referring to the raw `DynamoDBScanExpression<>` type.

Since we're testing the actual value of this argument with a captor later, would it be sufficient to use `any()` or `notNull()` instead? In general though, since Java's doing the type checking here, I'm not sure using the `isA(class)` matcher is adding value unless we care about the value being a specific subtype.

(This happens a couple of times in this file)

Reply:
how does java surface the error?

Reply:
In the [build output](https://travis-ci.org/blox/blox/builds/289718504#L581); before we had no warnings, now we have:

```
Note: /home/travis/build/blox/blox/data-service/src/test/java/com/amazonaws/blox/dataservice/repository/EnvironmentRepositoryDDBTest.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
```

Reply:
Thinking about this, we shouldn't have to go dig in build output for issues like these in PRs. Any objection to adding `-Wlint:all -Werror` to our `javac` compiler flags so that these issues actually become visible as build errors?